source/validate_id.cpp

```diff
@@ -779,14 +779,21 @@ bool idUsage::isValid<SpvOpLoad>(const spv_instruction_t* inst,
                        << "' is not a pointer.";
     return false;
   }
-  auto type = usedefs_.FindDef(pointer.second.words[1]);
-  assert(type.first);
-  spvCheck(resultType.second.id != type.second.id,
-           DIAG(resultTypeIndex)
-               << "OpLoad Result Type <id> '" << inst->words[resultTypeIndex]
-               << " does not match Pointer <id> '" << pointer.second.id
-               << "'s type.";
-           return false);
+  auto pointerType = usedefs_.FindDef(pointer.second.words[1]);
+  if (!pointerType.first || pointerType.second.opcode != SpvOpTypePointer) {
```

Comment:
This check for existence of the def and whether it's a pointer type suggests  a short utility function. 
The pattern is also used within OpVariable, OpCopyMemorySized.

Maybe we can do that later, along with a use of a DiagnosticStream object to shorten the return sequences.  But let's wait so we avoid making challenging merges for collaborators.